library/std/src/sys/uefi/common.rs

```diff
@@ -0,0 +1,477 @@
+        }
+        Status::IP_ADDRESS_CONFLICT => {
+            const_io_error!(ErrorKind::AddrInUse, "EFI_IP_ADDRESS_CONFLICT")
+        }
+        Status::LOAD_ERROR => {
+            const_io_error!(ErrorKind::Other, "EFI_LOAD_ERROR")
+        }
+        Status::MEDIA_CHANGED => {
+            const_io_error!(ErrorKind::StaleNetworkFileHandle, "EFI_MEDIA_CHANGED")
+        }
+        Status::NETWORK_UNREACHABLE => {
+            const_io_error!(ErrorKind::NetworkUnreachable, "EFI_NETWORK_UNREACHABLE")
+        }
+        Status::NO_MAPPING => {
+            const_io_error!(ErrorKind::Other, "EFI_NO_MAPPING")
+        }
+        Status::NO_MEDIA => {
+            const_io_error!(ErrorKind::Other, "EFI_NO_MEDIA")
+        }
+        Status::NO_RESPONSE => {
+            const_io_error!(ErrorKind::HostUnreachable, "EFI_NO_RESPONSE")
+        }
+        Status::NOT_FOUND => const_io_error!(ErrorKind::NotFound, "EFI_NOT_FOUND"),
+        Status::NOT_READY => const_io_error!(ErrorKind::ResourceBusy, "EFI_NOT_READY"),
+        Status::NOT_STARTED => const_io_error!(ErrorKind::Other, "EFI_NOT_STARTED"),
+        Status::OUT_OF_RESOURCES => {
+            const_io_error!(ErrorKind::OutOfMemory, "EFI_OUT_OF_RESOURCES")
+        }
+        Status::PROTOCOL_ERROR => {
+            const_io_error!(ErrorKind::Other, "EFI_PROTOCOL_ERROR")
+        }
+        Status::PROTOCOL_UNREACHABLE => {
+            const_io_error!(ErrorKind::Other, "EFI_PROTOCOL_UNREACHABLE")
+        }
+        Status::SECURITY_VIOLATION => {
+            const_io_error!(ErrorKind::PermissionDenied, "EFI_SECURITY_VIOLATION")
+        }
+        Status::TFTP_ERROR => const_io_error!(ErrorKind::Other, "EFI_TFTP_ERROR"),
+        Status::TIMEOUT => const_io_error!(ErrorKind::TimedOut, "EFI_TIMEOUT"),
+        Status::UNSUPPORTED => {
+            const_io_error!(ErrorKind::Unsupported, "EFI_UNSUPPORTED")
+        }
+        Status::VOLUME_FULL => {
+            const_io_error!(ErrorKind::StorageFull, "EFI_VOLUME_FULL")
+        }
+        Status::VOLUME_CORRUPTED => {
+            const_io_error!(ErrorKind::Other, "EFI_VOLUME_CORRUPTED")
+        }
+        Status::WRITE_PROTECTED => {
+            const_io_error!(ErrorKind::ReadOnlyFilesystem, "EFI_WRITE_PROTECTED")
+        }
+        _ => io::Error::new(ErrorKind::Uncategorized, format!("Status: {}", s.as_usize())),
+    }
+}
+
+pub fn install_protocol<T>(
+    handle: &mut r_efi::efi::Handle,
+    mut guid: r_efi::efi::Guid,
+    interface: &mut T,
+) -> io::Result<()> {
+    let boot_services = uefi::env::get_boot_services().ok_or(BOOT_SERVICES_ERROR)?;
+    let r = unsafe {
+        ((*boot_services.as_ptr()).install_protocol_interface)(
+            handle,
+            &mut guid,
+            r_efi::efi::NATIVE_INTERFACE,
+            interface as *mut T as *mut crate::ffi::c_void,
+        )
+    };
+    if r.is_error() { Err(status_to_io_error(r)) } else { Ok(()) }
+}
+
+pub fn uninstall_protocol<T>(
```

Comment:
You should probably add `pub(crate)` to all these functions just to make it clear they aren't exposed publicly. Making sure no API changes accidentally make it through this PR is one of the harder parts of this review, so I'd appreciate it.